boards/arm/qemu_cortex_r5/board.cmake

```diff
@@ -18,4 +18,12 @@ set(QEMU_KERNEL_OPTION
   "-device;loader,addr=0xff9a0000,data=0x80000218,data-len=4"
   )
 
-board_set_debugger_ifnset(qemu)
+board_runner_args(qemu "--commander=qemu-system-xilinx-aarch64" "--cpu=cortex-r5" "--machine=arm-generic-fdt"
+             "--dtb=${ZEPHYR_BASE}/boards/${ARCH}/${BOARD}/fdt-single_arch-zcu102-arm.dtb"
+             "--qemu_option=${ZEPHYR_BASE}/boards/${ARCH}/${BOARD}/qemu_cortex_r5_option.yaml")
+
+FILE(WRITE "${PROJECT_BINARY_DIR}/qemu_runner_variables.conf" "") # Empty file
+FILE(APPEND "${PROJECT_BINARY_DIR}/qemu_runner_variables.conf" ${${logical_target_for_zephyr_elf}}\n)
+FILE(APPEND "${PROJECT_BINARY_DIR}/qemu_runner_variables.conf" "-m ${QEMU_CPU_TYPE_${ARCH}}"\n)
```

Comment:
What is the purpose of this file?

Why can this not be supported using runner arguments to the QEMU runner?

Reply:
I put some the the options in qemu, but if keep adding options this way seems not good, so I decide to put new options to the qemu_runner_variables.conf file.

Reply:
>  if keep adding options this way seems not good

Why is this "not good"?

The current design keeps everything self-contained in the runners yaml file. This change adds an extra layer of indirection. If it is necessary please explain why. If it is personal preference I think that it would be better to keep things simple and self contained.

Reply:
@mbolivar-ampere  here is the current implement
```
board_runner_args(qemu "--commander=qemu-system-xilinx-aarch64" "--cpu=cortex-r5" "--machine=arm-generic-fdt"
             "--dtb=${CMAKE_CURRENT_LIST_DIR}/fdt-single_arch-zcu102-arm.dtb"
             "--qemu_option=${CMAKE_CURRENT_LIST_DIR}/qemu_cortex_r5_option.yaml")

include(${ZEPHYR_BASE}/boards/common/qemu.board.cmake)
```
so the qemu_cortex_r5_option.yaml contains below
```
qemu:
  qemu_option:
    - "-device loader,file={self.elf_name},cpu-num=4"
    - "-device loader,addr=0xff5e023c,data=0x80008fde,data-len=4"
    - "-device loader,addr=0xff9a0000,data=0x80000218,data-len=4"
    - "-net none -pidfile qemu.pid -chardev stdio,id=con,mux=on"
    - "-serial chardev:con -mon chardev=con,mode=readline"
    - "-icount shift=3,align=off,sleep=off -rtc clock=vm"
```

which is defined in board.cmake as
```
set(QEMU_KERNEL_OPTION
  "-device;loader,file=\$<TARGET_FILE:\${logical_target_for_zephyr_elf}>,cpu-num=4"
  "-device;loader,addr=0xff5e023c,data=0x80008fde,data-len=4"
  "-device;loader,addr=0xff9a0000,data=0x80000218,data-len=4"
  )
```
this setting pass to the Makefile, when build, but west can't get this QEMU_KERNEL_OPTION directly. the solution I can figure out is:
1. add options to board_runner_args(qemu, ...). but I find the $<TARGET_FILE:\${logical_target_for_zephyr_elf}, is not parased at build time, but reference from makefile when you run `make -t run`, so this need modified. which ends to I have to add another options, so I decide to add qemu related to a dedicateoption.yaml file if necessary.
2. parse the makefile when run west flash or west debug, this what I want to avoid.
Please let me know if there is a better solution. Thanks